scripts/prepare_transport_data_input.py

```diff
@@ -103,6 +103,13 @@ def download_CO2_emissions():
     # Drop region names that have no ISO2:
     CO2_emissions = CO2_emissions[CO2_emissions.country != "not found"]
 
+    # Drop region names where country column contains list of countries
```

Comment:
This is an interesting issue. What line fails?

I see it is linked to the code above:
```
    CO2_emissions = CO2_emissions.rename(columns={"Country Name": "Country"})
    cc = coco.CountryConverter()
    Country = pd.Series(CO2_emissions["Country"])
    CO2_emissions["country"] = cc.pandas_convert(
        series=Country, to="ISO2", not_found="not found"
    )
```
It is a bit surprising the problem you find as pandas_convert should thorugh "not found" when the conversion fails.
This code could be improved; for example, there is no need to re-create the pd.Series of a Series object.
Moreover, the use of CO2_emissions["country"] = serie may lead to unexpected behavior when index does not match.
This may be the reason of the issue you experience and lead to improper data matching.

If that is not sufficient, also discarding further unexpected values is clearly fine

Reply:
> This is an interesting issue. What line fails?
> 
> I see it is linked to the code above:
> 
> ```
>     CO2_emissions = CO2_emissions.rename(columns={"Country Name": "Country"})
>     cc = coco.CountryConverter()
>     Country = pd.Series(CO2_emissions["Country"])
>     CO2_emissions["country"] = cc.pandas_convert(
>         series=Country, to="ISO2", not_found="not found"
>     )
> ```
> 
> It is a bit surprising the problem you find as pandas_convert should thorugh "not found" when the conversion fails. This code could be improved; for example, there is no need to re-create the pd.Series of a Series object. Moreover, the use of CO2_emissions["country"] = serie may lead to unexpected behavior when index does not match; plus the format df[col] = ... may change behavior, conversely to the use f loc[:, col] These may be the reason of the issue you experience and lead to improper data matching.
> 
> If that is not sufficient, also discarding further unexpected values is clearly fine

Hi, @davide-f. Currently CIs are failing in `prepare_transport_data_input` rule. In particular, `country` column contains not only two letter codes (eg. `AU` or `DE`). But it also contains list `[IN, CN]`. It is because underlying data have such merged data. Here is the error:
![image](https://github.com/user-attachments/assets/b782ab93-aee5-4a24-b4fb-68073cc50555)

Reply:
> This is an interesting issue. What line fails?
> 
> I see it is linked to the code above:
> 
> ```
>     CO2_emissions = CO2_emissions.rename(columns={"Country Name": "Country"})
>     cc = coco.CountryConverter()
>     Country = pd.Series(CO2_emissions["Country"])
>     CO2_emissions["country"] = cc.pandas_convert(
>         series=Country, to="ISO2", not_found="not found"
>     )
> ```
> 
> It is a bit surprising the problem you find as pandas_convert should thorugh "not found" when the conversion fails. This code could be improved; for example, there is no need to re-create the pd.Series of a Series object. Moreover, the use of CO2_emissions["country"] = serie may lead to unexpected behavior when index does not match; plus the format df[col] = ... may change behavior, conversely to the use f loc[:, col] These may be the reason of the issue you experience and lead to improper data matching.
> 
> If that is not sufficient, also discarding further unexpected values is clearly fine

Conversions are fine, in my mind. The only thing is column may contain several countries. And it will return it as a list.

Reply:
> This is an interesting issue. What line fails?
> 
> I see it is linked to the code above:
> 
> ```
>     CO2_emissions = CO2_emissions.rename(columns={"Country Name": "Country"})
>     cc = coco.CountryConverter()
>     Country = pd.Series(CO2_emissions["Country"])
>     CO2_emissions["country"] = cc.pandas_convert(
>         series=Country, to="ISO2", not_found="not found"
>     )
> ```
> 
> It is a bit surprising the problem you find as pandas_convert should thorugh "not found" when the conversion fails. This code could be improved; for example, there is no need to re-create the pd.Series of a Series object. Moreover, the use of CO2_emissions["country"] = serie may lead to unexpected behavior when index does not match; plus the format df[col] = ... may change behavior, conversely to the use f loc[:, col] These may be the reason of the issue you experience and lead to improper data matching.
> 
> If that is not sufficient, also discarding further unexpected values is clearly fine

Hi, @davide-f. I have addressed your comments on removing redefinition of Series and using `.loc[:, col]`. Can you please check the PR? It used to be a simple PR to fix one of the current issues of CI and general workflow. Otherwise sector is does not work.